driver-core/src/main/com/mongodb/connection/ConnectionDescription.java

```diff
@@ -63,12 +64,14 @@ public ConnectionDescription(final ServerId serverId) {
      * @param maxBatchCount   the max batch count
      * @param maxDocumentSize the max document size in bytes
      * @param maxMessageSize  the max message size in bytes
+     * @deprecated Prefer {@link #ConnectionDescription(ConnectionId, ServerVersion, int, ServerType, int, int, int, List)}
      */
+    @Deprecated
```

Comment:
No reason that these constructors need to remain, as I don't expect any applications to create any.